core/src/main/scala/kafka/controller/ControllerContext.scala

```diff
@@ -391,6 +404,54 @@ class ControllerContext {
     partitionsForTopic(topic).filter { partition => states.contains(partitionState(partition)) }.toSet
   }
 
+  def putLeadershipInfo(partition: TopicPartition, leaderIsrAndControllerEpoch: LeaderIsrAndControllerEpoch): Unit = {
```

Comment:
Wonder if we should consider making `partitionLeadershipInfo` private so that we do not mistakenly invoke `put` in the future directly.

Reply:
That makes total sense. I have updated the PR to make it private. I have added methods in the controller context for all the reads (moved code around).